Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py

```diff
@@ -508,34 +529,56 @@ def module_test(client: Client):
 
 def fetch_indicators(client: Client, auto_sync: bool = False):
     if not get_integration_context() and auto_sync:
+        demisto.debug("running sync with first_time=True")
+        # this will happen on the first time we run
         xdr_iocs_sync_command(client, first_time=True)
     else:
+        # this will happen every minute
         get_changes(client)
         if auto_sync:
             tim_insert_jsons(client)
-            if iocs_to_keep_time():
+            demisto.debug("checking if iocs_to_keep should run")
+            if is_iocs_to_keep_time():
                 # first_time=False will call iocs_to_keep
+                demisto.debug("running sync with first_time=False")
                 xdr_iocs_sync_command(client)
 
 
 def xdr_iocs_sync_command(client: Client, first_time: bool = False):
     if first_time or not get_integration_context():
+        # the sync is the large operation including the data and the get_integration_context is fill in the sync
         sync(client)
     else:
         iocs_to_keep(client)
 
 
-def iocs_to_keep_time():
-    hour, minute = get_integration_context().get('iocs_to_keep_time', (0, 0))
-    time_now = datetime.now(timezone.utc)
-    return time_now.hour == hour and time_now.min == minute
-
-
-def create_iocs_to_keep_time():
+def set_new_iocs_to_keep_time():
     offset = secrets.randbelow(115)
-    hour, minute, = divmod(offset, 60)
+    hour, minute = divmod(offset, 60)
     hour += 1
-    return hour, minute
+    last_ioc_to_keep = datetime.now(timezone.utc)
+    last_ioc_to_keep = last_ioc_to_keep.replace(hour=hour, minute=minute) + timedelta(
+        days=1
+    )
+    next_iocs_to_keep_time = last_ioc_to_keep.strftime(DEMISTO_TIME_FORMAT)
+    demisto.debug(f"Setting next iocs to keep time to {next_iocs_to_keep_time}.")
+    set_integration_context(
+        get_integration_context()
```

Comment:
This will override the existing value of next_iocs_to_keep_time from the integration_context.